gensim/utils.py

```diff
@@ -266,6 +266,10 @@ def load(cls, fname, mmap=None):
         obj = unpickle(fname)
         obj._load_specials(fname, mmap, compress, subname)
         logger.info("loaded %s", fname)
+        if cls.__name__ == 'Similarity':
+            if obj.output_prefix is not fname:
+                obj.output_prefix = os.path.join(fname[: fname.rfind("shard")] , 'shard' ,'')   # '' to get trailing slash
```

Comment:
This deserves a comment -- why are we removing / adding the string `shard`?

Also nitpick: space after comma, not before.